goshopify.go

```diff
@@ -203,19 +203,11 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 		return err
 	}
 
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return err
-	}
-
 	if v != nil {
-		err := json.Unmarshal(bodyBytes, &v)
+		decoder := json.NewDecoder(resp.Body)
```

Comment:
How is a decoder different than Unmarshal? Is decoder the new standard?

Reply:
They're similar but can be used in slightly different situations. From Stack Overflow:

* Use `json.Decoder` if your data is coming from an `io.Reader` stream, or you need to decode multiple values from a stream of data.
* Use `json.Unmarshal` if you already have the JSON data in memory.

Reply:
Got it.